### ISSUE TITLE
ERR in importing old recordings from servers

### ISSUE THREAD

**Issue by rabser** [blindsidenetworks/scalelite#96]
With some recordings (not all of them) i'm getting this error:
```
E, [2020-03-24T23:39:20.807958 #6] ERROR -- : Failed to import recording: undefined method `update!' for nil:NilClass
W, [2020-03-24T23:39:20.808126 #6]  WARN -- : /srv/scalelite/lib/recording_importer.rb:31:in `block (2 levels) in import': undefined method `update!' for nil:NilClass (NoMethodError)

```
i would like to understand why, but i'm concerning where is defined the update! method ?
I can't find it in the code: it's a bug ?

Thanks a lot for your efforts, i'm going addicted with scalelite, but i'm not a ruby developer.

**Comment by rabser** [comment]
forget it for now, the metadata.xml it's in the wrong place and the internal xml seems different from the new recordings, i"ll investigate further.
Do you know anything about changes in storage format for BBB recordings  ?

**Comment by kurjajuur666** [comment]
Having a similar issue. My recordings are from 2016 and much older bigbluebutton.
Only difference being:
`WARN -- : /srv/scalelite/app/models/recording.rb:30:in `create_from_metadata_xml': undefined method `[]' for nil:NilClass (NoMethodError)`

**Comment by rabser** [comment]
This is because i tried to reformat the tar archive, trying to understand (working by difference with a new recording) how to modify the metadata.xml to be "compliant" with the new structure (missing fields, changed fields). But i didn't completely realize how to do these changes, for now.

**Comment by rabser** [comment]
I understood that the old recordings (older than 2017) had their metadata.xml different from the newer one in some tags format, but the info is there, so i wrote a simple python script to convert the scalelite tars which do not get acquired from scalelite-recording-importer to a new tar which works.
Rename the attached file to this post to convertRecordingsFromOldFormat.py, and exec as follow:

move tars from recording/spool directory somewhere
mkdir out
./convertRecordingsFromOldFormat.py --tarfile <somewhere/recordingid_taken_from_spool>.tar --outdir=./out

when finished, copy back the tar inside the outdir to the spool dir and wait for recording-importer to digest it...

[convertRecordingsFromOldFormat.txt](https://github.com/blindsidenetworks/scalelite/files/4424573/convertRecordingsFromOldFormat.txt)

It worked for me, hope this can help someone, but no warranty...

**Comment by SebastianZiegler** [comment]
Tried but some tags are not found

 File "./covertrecordings", line 46, in <module>
    recording['meta_context']=root.findall('./meta/context')[0].text
IndexError: list index out of range
 
Any idea?

**Comment by rabser** [comment]
try to comment out this tag catch and set it to empty string. My recordings were all done through Moodle, so maybe your recordings were done elsewhere and do not have this tag.
The only tags really important are id, name and a few others.

**Comment by SebastianZiegler** [comment]
in my case: playback url of some records was:
\n              https://xxxxxxxxxxx                \n
in xml file. After changing this import works.

**Comment by TheJJ** [comment]
Fixed with #163

**Comment by pielonet** [comment]
Hi,
I'm encountering the same error when trying to import already published recording from one Scalelite to the other :+1: 
```
E, [2020-10-22T14:18:30.457747 #7] ERROR -- : Failed to import recording: undefined method `update!' for nil:NilClass
W, [2020-10-22T14:18:30.458288 #7]  WARN -- : /srv/scalelite/lib/recording_importer.rb:31:in `block (2 levels) in import': undefined method `update!' for nil:NilClass (NoMethodError)
```
I have checked metadata.xml in both old and new recordings and they look quite the same. Both are created by Moodle.

What could be going wrong ?

Kind regards,
Thierry
